### PR TITLE
chore(tests): Cleanup Test21And26

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
I wanted to use the tests, but didn't feel the structure was ideal, and tests took 100ms longer than expected to execute (**each**), so I fixed them

## Approach
* Use `@Parameterized.Parameter`
* Use `@SuppressLint`
* Move the mocking to `.cctor`, rather than `.ctor`, 

## How Has This Been Tested?
It is tests

## Learning (optional, can help others)
See approach

* `abstract` stops a test class being attempted to be run

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by mikehardy — APPROVED** [review]
Interesting - yes cleaner on all the refactor nice

**Comment by david-allison** [comment]
Add another fix... break the tests. Today hasn't gone great 🤡

File: AnkiDroid/src/test/java/com/ichi2/compat/Test21And26.kt [hunk 1]
```diff
@@ -50,6 +46,14 @@ open class Test21And26(
         }.asIterable()
     }
 
+    @Parameterized.Parameter(0)
+    lateinit var compat: Compat
+
+    @Parameterized.Parameter(1)
+    @Suppress("unused")
+    /** Used in the "Test Results" Window */
+    lateinit var unitTestDescription: String
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/test/java/com/ichi2/compat/Test21And26.kt:70]: Thanks. I had no idea what I was doing (appart that it worked) and this first commit is really cleaner!

File: AnkiDroid/src/test/java/com/ichi2/compat/Test21And26.kt [hunk 2]
```diff
@@ -44,6 +45,20 @@ abstract class Test21And26 {
...
+        @BeforeClass
+        @JvmStatic
+        fun setupClass() {
+            mockkObject(CompatHelper)
+            every { CompatHelper.compat } answers { staticCompat }
+        }
+
+        @AfterClass
+        @JvmStatic
+        fun tearDown() {
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/test/java/com/ichi2/compat/Test21And26.kt:59]: I'd love `teardownClass`, to be consistent with `setupClass`

Comment (Arthur-Milchior) [inline AnkiDroid/src/test/java/com/ichi2/compat/Test21And26.kt:59]: I'd love `teardownClass`, to be consistent with `setupClass`

**Comment by mikehardy** [comment]
marked wait-for-feedback to give David a chance to re-push if he wants, otherwise anyone can merge I think

**Pull request merged**